lemur/models.py

```diff
@@ -0,0 +1,312 @@
+               ', experiments: {experiments}, classes: {classes}'
+               ', users: {users}>')
+        formatted = tpl.format(id=self.id, name=self.name,
+                               class_name=self.class_name,
+                               prof_name=self.prof_name,
+                               description=self.description,
+                               status=self.status,
+                               experiments=[e.name for e in self.experiments],
+                               classes=[c.name for c in self.classes],
+                               users=[u.name for u in self.users])
+        return formatted
+
+
+# Store experiments' info of a lab
+class Experiment(db.Model):
+    __tablename__ = 'Experiment'
+    id = db.Column(db.String(128), nullable=False, unique=True,
+                   primary_key=True)
+    name = db.Column(db.String(512), nullable=False)
+    description = db.Column(db.String(512))
+    order = db.Column(db.Integer, nullable=False)
+    # Type of value expected for this experiment
+    value_type = db.Column(db.Enum('Number', 'Text', name='value_type'))
+    # Value range only applies when the type is a number
+    value_range = db.Column(db.String(64))
+    # Value candidates only applies when the type is text
+    value_candidates = db.Column(db.String(512))
+
+    # Many-to-One: a lab can have multiple experiments
+    lab_id = db.Column(db.String(128), db.ForeignKey('Lab.id'))
+    lab = db.relationship("Lab", back_populates="experiments")
+
+    # one-to-many: a experiment can have multiple data
+    observations = db.relationship('Observation', back_populates='experiment',
+                                   cascade="all, delete, delete-orphan")
+
+    def __repr__(self):
+        tpl = ('Lab<id: {id}, name: {name}, description: {description}'
+               ', order: {order}, value_type: {value_type}'
+               ', value_range: {value_range}'
+               ', value_candidates: {value_candidates}'
+               ', lab_id: {lab_id}, lab:{lab}'
+               ', observations: {observations}>')
+        formatted = tpl.format(id=self.id, name=self.name,
+                               description=self.description,
+                               order=self.order,
+                               value_type=self.value_type,
+                               value_range=self.value_range,
+                               value_candidates=self.value_candidates,
+                               lab_id=self.lab_id,
+                               lab=self.lab.name,
+                               observations=[ob.id for ob in self.observations])
+        return formatted
+
+
+# Store data entered by students
+class Observation(db.Model, DateTimeInfo):
+    __tablename__ = 'Observation'
+    id = db.Column(db.String(128), nullable=False, unique=True,
+                   primary_key=True)
+    student_name = db.Column(db.String(256), nullable=False)
```

Comment:
Although we can unparse the name and map an Observation to several Students, the app seems doesn't need this mapping. Unless we want to add functions that enable the students to review the their history of data entry. This feature is not asked at this point since this app is only a collector that used to collect data and retrieve data. It is not used to keep track of all those data since the admin will put the cleaned data somewhere else.